nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -80,12 +80,16 @@ class ClusteringDiarizer(Model, DiarizationMixin):
     All the parameters are passed through config file 
     """
 
-    def __init__(self, cfg: DictConfig, speaker_model=None):
-        cfg = model_utils.convert_model_config_to_dict_config(cfg)
-        # Convert config to support Hydra 1.0+ instantiation
-        cfg = model_utils.maybe_update_config_version(cfg)
+    def __init__(self, cfg: Union[DictConfig, Any], speaker_model=None):
+        if isinstance(cfg, DictConfig):
+            cfg = model_utils.convert_model_config_to_dict_config(cfg)
+            # Convert config to support Hydra 1.0+ instantiation
+            cfg = model_utils.maybe_update_config_version(cfg)
         self._cfg = cfg
 
+        default_device = "cuda" if torch.cuda.is_available() else "cpu"
```

Comment:
Agree, might see how far we can get by just inheriting from `torch.nn.Module`

Reply:
I've mimicked `device` and `to` as now described in the README. turns out it will require quite a large refactor to make it fully support `device` and `to` as a torch.nn.Module. We should save this for a separate PR.

Reply:
Managed to get this working via LightningModule.